sdk/storage/AzureStorageBlob/Source/Credentials/StorageCredentials.swift

```diff
@@ -29,23 +29,74 @@ import Foundation
 
 /// A Storage shared access signature credential object.
 public struct StorageSASCredential: AzureCredential {
-    internal let blobEndpoint: String?
-    internal let queueEndpoint: String?
-    internal let fileEndpoint: String?
-    internal let tableEndpoint: String?
-    internal let sasToken: String?
-    internal let error: Error?
+    internal let credentialProvider: () throws -> String
+    internal var blobEndpoint: String?
+    internal var queueEndpoint: String?
+    internal var fileEndpoint: String?
+    internal var tableEndpoint: String?
+    internal var error: Error?
```

Comment:
Why are these changed from `var` to `let`?

Reply:
Whenever we need to refresh the token, we mutate these properties so they have to be `var`s and the functions `clear()` and `refresh()` have to be marked as `mutating`.